### DISCUSSION TITLE
Keyboard Translate Mode for Mac aarch64

### DISCUSSION THREAD

**Discussion by gaevoc** [rustdesk/rustdesk#7365] [General]
Hi all,
I am trying to use Rustdesk between a Mac M1 with an Italian keyboard and a RaspberryPi box. In this scenario the MacOS client only shows "Map mode" for keyboard but the keys sent to the controlled RasPi are not the right ones.
I read about a "Translate mode" but I can't find a way to set it, is it available on Mac?

Thank you
Gaetano

**Comment by GuillaumeHullin** [comment]
The Map client has translate mode. But I'm not sure about your Rasperry Pi client.

**Comment by gaevoc** [comment]
Hello Guillaume,
Thank you for your clarification.
As the “Translate” option doesn’t appear in my Mac client, does this mean that such option appears only when it is supported by both clients?

Gaetano

**Reply by fufesou** [reply]
Do you use Wayland on your RaspberryPi box?

**Reply by GuillaumeHullin** [reply]
@gaevoc it would make sense to hide what is not supported.

when you say:
> "Map mode" for keyboard but the keys sent to the controlled RasPi are not the right ones

Do you means that when you press a letter on your keyboard, it shows something completely off? or it would make sense if your keyboard was US layout?

**Reply by gaevoc** [reply]
The Mac has an Italian keyboard, the Raspberry is headless with Italian keyboard configured in the settings. When I press a key on the keyboard of the Mac I get a different character on the remote connection to the Raspberry. For example, the ':' is rendered as '>' on the screen.
I am pretty sure that there is X11 on the Raspberry, I will double check this

**Reply by GuillaumeHullin** [reply]
Alright but could it be that your Rasperry is in US layout?
If translate mode is not available, then you should be in map mode... which press the same "key" on the other side but the key is not necessarily the same character.

**Comment by gaevoc** [comment]
Exactly. As Map mode is not available the effect is the one you describe. On the raspberry the Italian keyboard is set. My question is why the Map mode is not available? I understand that it may not be available on the Raspberry side, how to confirm that?Il giorno 27 mar 2024, alle ore 20:54, Guillaume Hullin ***@***.***> ha scritto:﻿
Alright but could it be that your Rasperry is in US layout?
If translate mode is not available, then you should be in map mode... which press the same "key" on the other side but the key is not necessarily the same character.

**Comment by gaevoc** [comment]
There is one more thing that I just remember, if I connect to the raspberry pi from a windows machine, the Transpose mode is available. This means that the feature is missing on the Mac sideIl giorno 27 mar 2024, alle ore 22:59, Gaetano Vocca ***@***.***> ha scritto:﻿Exactly. As Map mode is not available the effect is the one you describe. On the raspberry the Italian keyboard is set. My question is why the Map mode is not available? I understand that it may not be available on the Raspberry side, how to confirm that?Il giorno 27 mar 2024, alle ore 20:54, Guillaume Hullin ***@***.***> ha scritto:﻿
Alright but could it be that your Rasperry is in US layout?
If translate mode is not available, then you should be in map mode... which press the same "key" on the other side but the key is not necessarily the same character.

**Comment by GuillaumeHullin** [comment]
Well I use a Mac M2 and I have the translate mode beta when I control Windows and Mac. Sorry I don't have a Linux with GUI to test.
